### PR TITLE
overlord/snapstate:  hold refreshes for 2h after seeding on classic

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: overlord/snapstate/snapmgr.go [hunk 1]
```diff
@@ -486,10 +486,30 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// atSeed implements at seeding policy for refreshes.
+func (m *SnapManager) atSeed() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
```
Comment (bboozzoo) [inline overlord/snapstate/snapmgr.go]: Should we set `seeded` once the operation is complete, or is it just for the tests?

Comment (pedronis) [reply overlord/snapstate/snapmgr.go]: that's done in devicestate

File: overlord/snapstate/autorefresh.go [hunk 2]
```diff
@@ -126,6 +128,30 @@ func (m *autoRefresh) clearRefreshHold() {
...
+		err := tr.Get("core", "refresh.hold", &t1)
+		if !config.IsNoOption(err) {
+			// already set or error
+			return err
+		}
+		// the 20mins tries to ensure that an autoRefresh.Ensure
+		// recalculating the refresh happens before the full 6h
+		// (default Ensure interval is 5mins)
+		maxDelta := 6*time.Hour - 20*time.Minute
+		delta := time.Duration(rand.Int63n(int64(maxDelta)))
```
Comment (niemeyer) [inline overlord/snapstate/autorefresh.go]: I thought we were going to delay for 6h, but this is random on 0-6h, which means we may not delay at all too. What am I missing?

File: overlord/snapstate/autorefresh.go [hunk 3]
```diff
@@ -126,6 +127,27 @@ func (m *autoRefresh) clearRefreshHold() {
...
+		tr := config.NewTransaction(m.state)
+		err := tr.Get("core", "refresh.hold", &t1)
+		if !config.IsNoOption(err) {
+			// already set or error
+			return err
+		}
+		// TODO: have a policy that if the snapd exe itself
+		// is older than X weeks/months we skip the holding?
+		now := time.Now().UTC()
+		tr.Set("core", "refresh.hold", now.Add(2*time.Hour))
```
Comment (mvo5) [inline overlord/snapstate/autorefresh.go]: Silly question: The PR title talks about 6h (and iirc we discussed 6h in the past). What is the background about changing it to 2h now?

Comment (pedronis) [reply overlord/snapstate/autorefresh.go]: it's the result of various back and forth, with the default schedule we will still pick something in a 6h window anyhow, but we hold for 2h now

**Comment by pedronis** [comment]
this depended on #4789

**Pull request merged**